modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,208 @@
+              created_at = Date.strptime(request[:created_date], '%m/%d/%Y')
+              next if created_at < 30.days.ago
+            end
+
+            klass = request.key?(:cc_appointment_request) ? CC : VA
+            model = build_appointment_model(request, klass)
+
+            if klass == VA
+              va_appointments << model
+            else
+              cc_appointments << model
+            end
+          end
+
+          [va_appointments, cc_appointments]
+        end
+
+        private
+
+        # rubocop:disable Metrics/MethodLength
+        def build_appointment_model(request, klass)
+          start_date = localized_start_date(request)
+
+          Mobile::V0::Appointment.new(
+            id: request[:appointment_request_id],
+            appointment_type: klass::APPOINTMENT_TYPE,
+            cancel_id: nil,
+            comment: nil,
+            facility_id: klass.facility_id(request),
+            sta6aid: nil,
+            healthcare_provider: klass.provider_name(request),
+            healthcare_service: klass.practice_name(request),
+            location: klass.location(request),
+            minutes_duration: nil,
+            phone_only: phone_only?(request),
+            start_date_local: start_date,
+            start_date_utc: start_date.utc,
+            status: status(request),
+            status_detail: nil,
+            time_zone: time_zone(request),
+            vetext_id: nil,
+            reason: request[:purpose_of_visit],
+            is_covid_vaccine: nil, # unable to find or create test data for this
+            proposed_times: proposed_times(request),
+            type_of_care: request[:appointment_type],
+            visit_type: request[:visit_type],
+            patient_phone_number: request[:phone_number],
+            patient_email: request[:email],
+            best_time_to_call: request[:best_timeto_call]
+          )
+        end
+        # rubcop:enable Metrics/MethodLength
+
+        def phone_only?(request)
+          return false if request[:text_messaging_allowed].nil?
+
+          !request[:text_messaging_allowed]
+        end
+
+        def proposed_times(request)
+          {
+            option_date1: request[:option_date1],
+            option_time1: request[:option_time1],
+            option_date2: request[:option_date2],
+            option_time2: request[:option_time2],
+            option_date3: request[:option_date3],
+            option_time3: request[:option_time3]
+          }
+        end
+
+        def localized_start_date(request)
+          date = request[:option_date1]
+          time_zone = time_zone(request) || '+00:00'
+          month, day, year = date.split('/').map(&:to_i)
+          hour = request[:option_time1] == 'AM' ? 8 : 12
+
+          DateTime.new(year, month, day, hour, 0).in_time_zone(time_zone)
+        end
+
+        def status(request)
+          request[:status].upcase
+        end
+
+        # this is not correct for cc appointment requests.
```

Comment:
For the time being, I'm just using the facility data to determine time zone for all appointments. That should be mostly correct and is unlikely to be off by more than an hour. We could try finding a ruby library that converts zip codes to timezones.

Reply:
Yeah, that's about as good as we can get right now. Gets even more complicated when the parent facility and the clinic don't share a timezone. We have a blocked ticket that's wrapped up in all this stuff. Waiting on info from Lighthouse that might help us figure out how to proceed.

Reply:
I don't think zip codes are enough to tell you which time zone you're in. Lat Long can and we could pull down all the clinics and create a lookup table using the [Google maps API](https://developers.google.com/maps/documentation/timezone/get-started).

Reply:
Sounds like this is good enough for now then, because that definitely sounds like a separate ticket.